databricks/koalas/internal.py

```diff
@@ -394,6 +408,80 @@ def __init__(self, sdf: spark.DataFrame,
         else:
             self._column_index_names = column_index_names
 
+    @staticmethod
+    def attach_default_index(sdf):
+        """
+        This method attaches a default index to Spark DataFrame. Spark does not have the index
+        notion so corresponding column should be generated.
+
+        There are three types of default index that can be controlled by `DEFAULT_INDEX`
+        environment variable.
+
+        - one-by-one: It implements an one-by-one sequence by Window function without
+            specifying partition. Therefore, it ends up with whole partition in single node.
+            This index type should be avoided when the data is large. This is default.
+
+        - distributed-one-by-one: It implements an one-by-one sequence by group-by and
+            group-map approach. It still generates a one-by-one sequential index globally.
+            If the default index must be an one-by-one sequence in a large dataset, this
+            index has to be used.
```

Comment:
Maybe we should notice that, if we add more data to the data source after creating this index, then it doesn't guarantee the sequential index?